src/runtime/virtcontainers/sandbox.go

```diff
@@ -1623,6 +1623,11 @@ func (s *Sandbox) Stop(force bool) error {
 		return err
 	}
 
+	// Stop communicating with the agent.
+	if err := s.agent.disconnect(); err != nil {
```

Comment:
Please, add `&& !force` here.

IMHO, the error should only be reported in case `force` is `false`.